src/libcore/slice/mod.rs

```diff
@@ -1199,12 +1199,14 @@ macro_rules! iterator {
             fn position<F>(&mut self, mut predicate: F) -> Option<usize>
                 where F: FnMut(Self::Item) -> bool,
             {
+                let len = self.len();
                 let mut index = 0;
                 self.search_while(None, move |elt| {
                     if predicate(elt) {
                         SearchWhile::Done(Some(index))
                     } else {
                         index += 1;
+                        unsafe { assume(index < len); }
```

Comment:
(This one should be before += 1 to be correct.)

Reply:
Good point.

Reply:
@arthurprs  What if you make it `assume(0 < index && index <= len)`

Reply:
We can try that too. Lets try a few more things.